File: sdk/elixir/lib/telecom_sdk/rating_plans_api.ex
defmodule TelecomSDK.RatingPlanAPI do
  @moduledoc """
  API for rating plan management
  """

  defstruct [:client]

  def new(client) do
    %__MODULE__{client: client}
  end

  def list(api) do
    TelecomSDK.HTTPClient.get(api.client, "/v1/rating-plans")
  end

  def get(api, plan_id) do
    TelecomSDK.HTTPClient.get(api.client, "/v1/rating-plans/#{plan_id}")
  end
end

File: sdk/ruby/lib/telecom_sdk/rating_plans_api.rb
module TelecomSDK
  # API for rating plan management
  class RatingPlanAPI
    def initialize(client)
      @client = client
    end

    def list
      @client.get("/v1/rating-plans")
    end

    def get(plan_id)
      @client.get("/v1/rating-plans/#{plan_id}")
    end
  end
end
